Add "oldmembers" mailinglist

ColonelPhantom opened the pull request:
Closes #2773 

<!--

Please add the appropriate label for the change that you made to this PR:
- feature: new feature for the user, not a new feature for build script
- bug: fix for the user, not a fix to a build script
- docs: changes to the documentation
- refactor: refactoring production code, eg. renaming a variable or rewriting a function
- test: adding missing tests, refactoring tests; no production code change
- chore: updating poetry, changing the CI settings etc; no production code change

-->

### Summary
This adds a mailing list targeting alumni as requested. For now it is a draft because the PR is not fully complete, not tested at all, and the exact requirements are not 100% clear. @ricotewechel please let me know if the following logic matches what you want or not:
```python
m.current_membership.type != Membership.MEMBER and m.has_been_member()
```
where this condition is evaluated on all members `m`. This covers everyone who has once been a member, but is not anymore, for example users who go member->benefactor or member->honorary, or just member->nothing.
### How to test
NOT DONE YET

JobDoesburg commented on website/mailinglists/services.py:
```diff
@@ -86,6 +93,14 @@ def get_automatic_lists():
             "moderated": True,
         },
+        {
+            "name": "alumni",
```
the alias `alumni@thalia.nu` already exists and is used by the alumni committee. I would call the list `oldmembers`

JobDoesburg commented:
> ```python
> if m.current_membership_type == Membership.HONORARY
>     or (m.has_been_member() and not m.has_active_membership())
> ```
> 
> where this condition is evaluated on all members `m`. I believe that `has_been_member` does include people who go like member->benefactor>ex-member.
> 


I've discussed it with rico and it should indeed include only people that have ever been a `member` (so not just benefactor, but member → benefactor is fine). The current members should be filtered out, but the current benefactors/honorarymembers that have been a member previously, should be included. 

Finally, we should also make a opt-out for this, but that's for a later issue I think

ColonelPhantom commented on website/mailinglists/services.py:
```diff
@@ -55,6 +55,12 @@ def get_automatic_lists():
+    oldmembers = [
+        m
+        for m in Member.objects.all()
+        if m.current_membership.type != Membership.MEMBER and m.has_been_member()
```
As Job mentioned, this will likely throw an error when there are Members without a current_membership.

ColonelPhantom commented:
I have updated with a fix for the non-member issue, and added an opt-out flag to the profile.

We may want to hide the flag for current members (?) and/or enable people to set the flag without logging in (if they lost their credentials; although the Forgot Password button should be linked to the same address receiving the mails, so everyone who receives the emails should be able to access their account.)

JobDoesburg reviewed (approved):
This does still need a view for easy deregistration if you don't know your password anymore

ColonelPhantom commented:
> This does still need a view for easy deregistration if you don't know your password anymore

I'm still not sure if this is actually necessary; the forgot password functionality is inherently linked to the same email where you receive the emails. So if you can read the emails, you can reset your password and then unsubscribe. It's a bit more work but should be fairly discoverable, especially if the `edit-profile` link is included (click -> see login page -> forgot password -> enter email) in the emails (although I presume we can't really automate that, nor rely on it being included manually).

That said, less hurdles for unsubscribing is of course better. But I'm not entirely convinced the current barrier is too high, either.

JobDoesburg commented:
Sure let's keep it like this for now
